### PR TITLE
Update JSON schema validations of Daily Round fields 

### LINKED ISSUE #1775: Update validation for Daily Round form
**Describe the bug**
Currently there are lots of validations in daily rounds FE that are not there in BE.

Such as, validation to prevent empty log updates from being created.

- [ ] Add proper serializer validations for daily rounds.

### LINKED ISSUE #2368: Add more validations for json fields in daily round object
Add more validations for json fields in daily round object, and cleanup unsanitized data

### PR DESCRIPTION
**Pull request by rithviknishad**
> [!NOTE]
> PR contains **2 custom migrations** to sanitize daily rounds JSON fields
## Proposed Changes

- [x] Adds proper enum choices for all possible fields instead of strings.
- [x] Adds proper numeric ranges for all possible fields. 
- [x] Removed unused fields (PR: #708; issue: https://github.com/coronasafe/care_fe/issues/1980)
- [x] Migrate JSON fields with `name` with value: `<empty string>` to `"Unknown"`
- [x] Set blood pressure column value with `{}` to `None` to simplify validations.

### Associated Issue

- Closes #2368
- Closes #1775
- https://github.com/ohcnetwork/care_fe/issues/8397
- FE: https://github.com/ohcnetwork/care_fe/pull/8498

### Custom Migrations 1: Sanitize I/O Balance Fields

Migration to sanitize daily rounds I/O balance JSON fields.

#### Before Migration

image

#### After Migration

image

#### Reverse Migration

image

### Custom Migrations 2: Sanitize `bp` field

Migration to set `bp` field to `None` for `{}` (empty objects)

#### Before Migration

image

#### After Migration

image

#### Reverse Migration

image

## Merge Checklist

- [x] Tests added/fixed
- [x] Linting Complete
- [x] Any other necessary step

### CODE REVIEW

File: care/facility/api/serializers/daily_round.py
Comment (sainak) [inline care/facility/api/serializers/daily_round.py]: this could be an issue, the log updates filed by the middleware does not contain the mean value

what would be the default in case of normal log updates?

Comment (sainak) [reply care/facility/api/serializers/daily_round.py]: MAP = 1/3 × SBP + 2/3 × DBP

**Review by sainak — APPROVED** [review]

File: care/facility/api/serializers/daily_round.py
Comment (rithviknishad) [reply care/facility/api/serializers/daily_round.py]: @sainak Updated the schema to disallow `mean` from being stored to DB, and simplified validations too.

**Review by sainak — APPROVED** [review]

**Review by sainak — APPROVED** [review]

File: care/facility/api/serializers/daily_round.py
Comment (sainak) [inline care/facility/api/serializers/daily_round.py]:
```suggestion
        if value is not None:
            sys, dia = value.get("systolic"), value.get("diastolic")
            if sys is not None and dia is not None and sys < dia:
                raise ValidationError("Systolic must be greater than diastolic")
        return value
```

File: care/facility/models/json_schema/daily_round.py
Comment (rithviknishad) [inline care/facility/models/json_schema/daily_round.py]: Waiting for updated new set of procedures to be confirmed in https://github.com/ohcnetwork/care_fe/pull/8568/files#diff-37705b30476b14631124ce42dd5c3500d43fbede16f0e91f9ae2905e4adc6376R737-R766

**Comment by bodhish** [comment]
@anroopak can you review this PR?

File: care/facility/models/json_schema/daily_round.py
Comment (anroopak) [inline care/facility/models/json_schema/daily_round.py]: `Nor-adrenalin` is this a typo? or is it "No**n**-adrenalin"

**Review by anroopak — COMMENTED** [review]
Code LGTM, please confirm about if it is a typo or not before merging. Will approve after the confirmation. 

Custom migrations may cause the release to stall if the table is huge. So need confirmation that it is not so huge.

File: care/facility/models/json_schema/daily_round.py
Comment (rithviknishad) [reply care/facility/models/json_schema/daily_round.py]: @aparnacoronasafe could you confirm?

image

Comment (aparnacoronasafe) [reply care/facility/models/json_schema/daily_round.py]: It should be "Noradrenaline". The "-" is not needed

**Review by anroopak — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Pull request merged**
